eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.yegor256.tojos.MnJson;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.Map;
+import org.cactoos.Input;
+import org.cactoos.io.ResourceOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+/**
+ * Test case for {@link DiscoverMojo}.
+ *
+ * @since 0.28.11
+ */
+final class DiscoverMojoTest {
+
+    /**
+     * The name of foreign dependencies file.
+     */
+    private static final String EO_FOREIGN = "eo-foreign.json";
+
+    @ParameterizedTest
+    @CsvSource({
+        "org/eolang/maven/mess.eo, 7",
+        "org/eolang/maven/sum.eo, 0",
+        "org/eolang/maven/withwarning.eo, 1"
+    })
+    void executeDiscoveryPhaseForEoCorrectPrograms(
+        final String program,
+        final int dependencies,
+        final @TempDir Path tmp
+    ) throws IOException {
+        this.saveProgram(tmp, new ResourceOf(program));
+        this.discover(tmp);
+        final Deque<Map<String, String>> json = this.discoveredJsonEntries(tmp);
+        final Map<String, String> first = json.removeFirst();
+        MatcherAssert.assertThat(
+            String.valueOf(dependencies),
+            Matchers.equalTo(first.get("discovered"))
+        );
+        MatcherAssert.assertThat(dependencies, Matchers.equalTo(json.size()));
+    }
+
+    /*
+     * @todo #1360:30min Make logger inside DiscoveryMojo non-static.
```

Comment:
@volodya-lombrozo How is logger related to DiscoverMojo functionality?

Reply:
I can't mock it in order to write test with true check

Reply:
@volodya-lombrozo why would you need to mock a logger for tests? Can't it be set up appropriately for tests if needed?

Reply:
@mximp What do you mean? Do we have any other way to ask the logger about his last calls? Can you provide an example please? it's not so clear for me what is "set uo appropriately".

Reply:
@volodya-lombrozo I also feel that logging functionality is something orthogonal to object discovery functionality. We should not test logging in this test class, I believe. Also, it would be impossible to "inject logger as a field" since we use `com.jcabi.log.Logger` utility class everywhere, which redirects all calls to Maven logging facility.

Reply:
@yegor256 @mximp I don't wanna test the logger behavior, I wanna test `DiscoveryMojo` corner cases and how it behaves in different situations. It can be useful to see in tests a full picture. I often use tests as a class documentation and in tests I can see how the class actually works. If we have only successful paths in tests, I don't see the another important part and I have to open the class and read the full class code in order to understand his behaviour in corner cases. Moreover it increases test coverage.

Ok, for now I don't see any possibility to test it. I will remove the pazzle and disabled tests

Reply:
Done